stingray/gti.py

```diff
@@ -116,10 +116,9 @@ def create_gti_mask(time, gtis, safe_interval=0, min_length=0,
 
     check_gtis(gtis)
 
-    dt = assign_value_if_none(dt,
-                              np.zeros_like(time) +
-                              np.median(np.diff(time)) / 2)
+    dt_array = np.diff(np.sort(time))
```

Comment:
The previous operation only executed the diff if dt was None, while now it gets executed always. Can we put the sort operation inside the assign_value_if_none?